File: 554_brick_wall/leastBricks.ts
function leastBricks(wall: number[][]): number {
  const intersections = {};

  for (let i = 0; i < wall.length; i++) {
    let brick = 0;
    for (let j = 0; j < wall[i].length - 1; j++) {
      brick += wall[i][j];
      if (brick in intersections) {
        intersections[brick]++;
      } else {
        intersections[brick] = 1;
      }
    }
  }

  let max = 0;

  for (let i in intersections) {
    if (intersections[i] > max) {
      max = intersections[i];
    }
  }

  return wall.length - max;
}

File: 554_brick_wall/leastBricks.py
from typing import List


class Solution:
    def leastBricks(self, wall: List[List[int]]) -> int:
        map = {}

        for i in range(len(wall)):
            bricks = 0
            for j in range(len(wall[i]) - 1):
                bricks += wall[i][j]
                value = map.get(bricks, 0)
                map[bricks] = value + 1

        mx = 0

        for i in map.values():
            if i > mx:
                mx = i

        return len(wall) - mx
